provisioning/src/users_and_groups.js

```diff
@@ -3,24 +3,49 @@ const {
   createUser,
   createGroup,
   addUserToGroup,
-  removeUserFromGroup,
   grantAllPermissionsToUser,
+  grantGlobalPermissionToUser,
 } = require("./api");
 const log = require("./logger");
 
+const allowedIntents = [
+  "global.listPermissions",
+  "global.grantPermission",
+  "global.grantAllPermissions",
+  "global.revokePermission",
+  "global.createProject",
+  "global.createUser",
+  "global.enableUser",
+  "global.disableUser",
+  "global.listAssignments",
+  "global.createGroup",
+  "network.registerNode",
+  "network.list",
+  "network.listActive",
+  "network.voteForPermission",
+  "network.approveNewOrganization",
+  "network.approveNewNodeForExistingOrganization",
+  "network.declineNode",
+  "provisioning.start",
+  "provisioning.end",
+  "provisioning.get"
+];
+
 const provisionUsers = async (axios, folder, organization) => {
   try {
     const users = readJsonFile(folder + "users.json");
-    for (const user of users) {
-      log.info(`~> Adding user ${user.displayName}`);
-      await createUser(axios, user, organization);
-    }
-
-    await grantAllPermissionsToUser(axios, "mstein");
-    log.info("~> all Permissions granted to mstein");
 
-    await grantAllPermissionsToUser(axios, "jdoe");
-    log.info("~> all Permissions granted to jdoe");
+    await Promise.allSettled(users.map((user) => { log.info(`~> Adding user ${user.displayName}`); return createUser(axios, {id: user.id, displayName: user.displayName, password: user.password}, organization);}));
+    const usersWithPermissions = users.filter((user) => Array.isArray(user.permissions) && user.permissions.length > 0);
+    for (const user of usersWithPermissions) {
+      if (user.permissions.includes("all")) {
+        log.info(`~> all Permissions granted to ${user.id}`);
+        await grantAllPermissionsToUser(axios, user.id);
```

Comment:
what happens if the permission can not be granted? Maybe add some error handling?